warehouse/core/src/main/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormat.java

```diff
@@ -0,0 +1,137 @@
+                throw new IllegalArgumentException("Must have file splits");
+            }
+
+            // for each file get the index blocks
+            FileSplit fileSplit = (FileSplit) inputSplit;
+            log.info("creating splits for: " + fileSplit);
+            rfileSplits.addAll(getSplits(config, fileSplit));
+        }
+        log.info("total splits: " + rfileSplits.size());
+
+        return rfileSplits;
+    }
+
+    /**
+     * Create rfile splits for a given FileSplit. Splits will honor the MIN_BLOCKS_PER_SPLIT config option. Splits may extend more than MIN_BLOCKS_PER_SPLIT if
+     * a given Key spans more index blocks.
+     *
+     * @param config
+     * @param fileSplit
+     * @return
+     * @throws IOException
+     */
+    public static List<InputSplit> getSplits(Configuration config, FileSplit fileSplit) throws IOException {
+        log.info("getting splits for: " + fileSplit);
+        Path rfile = fileSplit.getPath();
+        FileSystem fs = rfile.getFileSystem(config);
+
+        if (!fs.exists(rfile)) {
+            throw new IllegalArgumentException(rfile + " does not exist");
+        }
+
+        CryptoService cs = CryptoFactoryLoader.getServiceForClient(CryptoEnvironment.Scope.TABLE, config.getPropsWithPrefix(TABLE_CRYPTO_PREFIX.name()));
+        CachableBlockFile.CachableBuilder cb = new CachableBlockFile.CachableBuilder().fsPath(fs, rfile).conf(config).cryptoService(cs);
+        RFile.Reader rfileReader = new RFile.Reader(cb);
+
+        // get the first and last keys to bound the blocks while creating splits
+        Key firstKey = rfileReader.getFirstKey();
+        Key lastKey = rfileReader.getLastKey();
+
+        // use the index blocks to create the splits
+        FileSKVIterator iter = rfileReader.getIndex();
+
+        // track the last split key to since multiple splits with the same split key MUST be in the same block
+        Key lastSplit = firstKey;
+        List<InputSplit> splits = new ArrayList<>();
+        int blkCount = 0;
+        int splitBlocks = 0;
+
+        int minBlocksPerSplit = config.getInt(MIN_BLOCKS_PER_SPLIT, 1);
+        log.info("Blocks per split: " + minBlocksPerSplit);
+
+        Key top = null;
+        while (iter.hasTop()) {
+            splitBlocks++;
+            top = iter.getTopKey();
+
+            if (!top.equals(lastSplit) && splitBlocks >= minBlocksPerSplit) {
+                splits.add(new RFileSplit(fileSplit.getPath(), fileSplit.getStart(), fileSplit.getLength(), fileSplit.getLocations(), blkCount, splitBlocks,
+                                top));
+                blkCount += splitBlocks;
+                lastSplit = iter.getTopKey();
+                splitBlocks = 0;
+            }
+            iter.next();
+        }
+
+        // if there is a gap between the last index key top key and the lastKey in the file or there hasn't been a split created yet
+        if (top.equals(lastKey) || splits.isEmpty()) {
```

Comment:
The comment mentions "if there is a gap..." but the first condition is checking if they are equal. Should the check be for "not equal"?

Reply:
good catch, I'll add a test for this case as well

Reply:
After going back to this, the conditional is actually not needed at all. In all cases there is a final split to add.